calculator.rb

```diff
@@ -0,0 +1,57 @@
+# Welcome message for user
+puts "Welcome to the Ada Calculator!"
+puts "These are the following choices:"
+puts "1. add(+)"
+puts "2. subtract(-)"
+puts "3. multiply(*)"
+puts "4. divide(/)"
+puts "Please choose an operand:"
+
+# Accepting user choice (case insensitive)
+operand = gets.chomp.downcase
+
+# Ensuring user is choosing a valid operation
+until ["add", "+", "subtract", "-", "multiply", "*", "divide", "/"].include? (operand) do
+  puts "Please enter a valid operand"
+  operand = gets.chomp.downcase
+end
+
+# Asking user to input values to be used in calculations
+puts "Please enter the first value: "
+first_value = gets.chomp.to_f
+
+# Insert logic to ensure first_value entered is a number
+
+puts "Please enter the second value: "
+second_value = gets.chomp.to_f
+
+# Insert logic to ensure second_value entered is a number
+
+# Defining methods for each operand
+def addittion(x, y)
+  return (x + y)
+end
+
+def subtraction(x, y)
+  return (x - y).to_f
```

Comment:
you end up calling `.to_f` on this result (and in some other calculation methods).

However, let's establish some things:
- in lines 21 and 26, you get the values for `first_value` and `second_value` by getting user input and then converting them to floats immediately with `.to_f`.

And let's remember:
- in Ruby, if you do addition, subtraction, multiplication, or division with any floats, the result will be a float

Therefore: do you need to convert to a float here?

Sometimes, it's good to have a `.to_f` in here just in case there's a chance you still need to convert it, but within this particular program, you probably wouldn't need to call `.to_f` here. Not a big deal! Just wanted to call out this redundancy.